src/DataSource/Http.hs

```diff
@@ -132,6 +144,26 @@ iterateTicketPages req = do
         Just nextUrl -> liftIO $ go page0 nextUrl
         Nothing      -> pure page0
 
+iterateUserPages
```

Comment:
If this is copy and paste, can you instead just genetalize the signature to somethin more general?

Reply:
I've tried but it's a bit difficult for me to do it.. I'll tackle again tomorrow morning.

Reply:
What about something like:
```
iteratePages
    :: forall m a. (MonadIO m, MonadReader Config m, FromJSON a)
    => Request -> m [a]
iteratePages req = do ...
```
Use `parseJSON` instead of `parseUsers` and it's counterpart, and create a general structure that contains the page and results:
```
data PageResultList a = PageResultList
    { tlResults :: ![a]
    , nextPage  :: Maybe Text -- Text ?
    }
```
Try it and if you get stuck, ping me.